### ISSUE TITLE
[Bug] 100% CPU loop when zoomed far in

### ISSUE THREAD

**Issue by pasniak** [osmdroid/osmdroid#1106]
When map is zoomed beyond the provider's zoom ability (?), SimpleInvalidationHandler continuously sends invalidate() leading to 100% CPU usage.  

I could repro this in the sample app; start it in Profiler, take any map and continue zooming on some meaningfull location, say New York downtown, Wall Street. After a while you will hit zoom level with constant 100% CPU in Profiler (where you see Trinity Church full screen)

![screenshot from 2018-07-21 08-30-04](https://user-images.githubusercontent.com/19778666/43035582-7412de4a-8cc0-11e8-9b4e-1beb38aeaaf3.png)

When I debug my app, I see SimpleInvalidationHandler
```
	public void handleMessage(final Message msg) {
		switch (msg.what) {
		case MapTileProviderBase.MAPTILE_SUCCESS_ID:
			if (mView!=null)
				mView.invalidate();
			break;
		}
	}
```
flooding the invalidations. Here's the log:
https://gist.github.com/pasniak/10e5863ded98a21cf132c3e21c36a9d8

You'd say: do not zoom so far... yet I want to zoom further to show overlayed images, which are at smaller scale than the available tiles...

**Comment by monsieurtanuki** [comment]
Just found an elegant solution.
I may PR tonight.

**Comment by monsieurtanuki** [comment]
Just PR'ed #1107
@pasniak Could you test your issue with this fix?

**Comment by pasniak** [comment]
I meant 100% core - this can drain the phone overnight if left zoomed in... :-) Let me try tonight, or, if not, tomorrow.

**Comment by pasniak** [comment]
Tested. Works great (in emu and my !  I upgraded from my frozen master as of Apr 2018 and all seems in order.

**Comment by spyhunter99** [comment]
great, i'll merge and hopefully cut a new release tonight

**Comment by spyhunter99** [comment]
https://github.com/osmdroid/osmdroid/issues/131 related

**Comment by monsieurtanuki** [comment]
@pasniak I have the impression that the recent code merges fix at the same time this issue, #131 and #989. Ready to close them all?
